src/fhost_fpga.py

```diff
@@ -303,6 +283,199 @@ def get_pfb_snapshot(self, pol=0):
+		delay_delta_reg = self.registers.delay_delta0
+		phase_offset_reg = self.registers.phase_offset0
+		phase_offset_delta_reg = self.registers.phase_offset_delta0
+	else:
+		fd_status_reg = self.registers.tl_fd1_status
+		fd_control_reg = self.registers.tl_fd1_control
+		fd_control0_reg = self.registers.tl_fd1_control0
+		cd_status_reg = self.registers.tl_cd1_status
+		cd_control_reg = self.registers.tl_cd1_control
+		cd_control0_reg = self.registers.tl_cd1_control0
+		coarse_delay_reg = self.registers.coarse_delay1
+		fractional_delay_reg = self.registers.fractional_delay1
+		delay_delta_reg = self.registers.delay_delta1
+		phase_offset_reg = self.registers.phase_offset1
+		phase_offset_delta_reg = self.registers.phase_offset_delta1
+
+        fine_delay_bits =       16
+        coarse_delay_bits =     16
+        fine_delay_rate_bits =  16
+        fringe_offset_bits =    16
+        fringe_rate_bits =      16
+        bitshift_schedule =     23
+        min_ld_time = 0.1 # assume we're able to set and check all the registers in 100ms
+        network_latency_adjust = 0.015
+
+        # delays in terms of ADC clock cycles:
+        delay_n = delay * self.sample_rate_hz                   # delay in clock cycles
+        coarse_delay = int(delay_n)                             # delay in whole clock cycles #testing for rev370
+        fine_delay = (delay_n-coarse_delay)                     # delay remainder. need a negative slope for positive delay
+        fine_delay_i = int(fine_delay*(2**(fine_delay_bits)))   # 16 bits of signed data over range 0 to +pi
+
+        fine_delay_rate = int(float(delay_rate) * (2**(bitshift_schedule + fine_delay_rate_bits-1)))
+
+        # figure out the fringe as a fraction of a cycle
+        fr_offset = int(fringe_phase/float(360) * (2**(fringe_offset_bits)))
+        # figure out the fringe rate. Input is in cycles per second (Hz). 1) divide by brd clock rate to get cycles per clock. 2) multiply by 2**20
+	# TODO adc_demux_factor
+	feng_clk = (self.sample_rate_hz/8)
+        fr_rate = int(float(fringe_rate) / feng_clk * (2**(bitshift_schedule + fringe_rate_bits-1)))
+
+        # read the arm and load counts - they must increment after the delay has been loaded
+	status = fd_status_reg.read()
+        arm_count_before = status['arm_count']
+        ld_count_before = status['load_count']
+
+        act_delay = (coarse_delay + float(fine_delay_i)/2**fine_delay_bits)/self.sample_rate_hz
+        act_fringe_offset = float(fr_offset)/(2**fringe_offset_bits)*360
+        act_fringe_rate = float(fr_rate)/(2**(fringe_rate_bits+bitshift_schedule-1))*feng_clk
+        act_delay_rate = float(fine_delay_rate)/(2**(bitshift_schedule + fine_delay_rate_bits-1))
+# TODO 
+        if (delay != 0):
+            if (fine_delay_i == 0) and (coarse_delay == 0):
+                LOGGER.info('Requested delay is too small for this configuration (our resolution is too low). Setting delay to zero.')
```

Comment:
Some very long lines here and below. I'll merge anyway, but would be nice to fix :)